Source/WebKit/WebProcess/Extensions/API/Cocoa/WebExtensionAPITabsCocoa.mm

```diff
@@ -428,6 +431,28 @@
     return true;
 }
 
+bool WebExtensionAPITabs::parseSendMessageOptions(NSDictionary *options, std::optional<WebExtensionFrameIdentifier>& frameIdentifier, NSString *sourceKey, NSString **outExceptionString)
+{
+    static NSDictionary<NSString *, id> *types = @{
+        frameIdKey: NSNumber.class,
+    };
+
+    if (!validateDictionary(options, sourceKey, nil, types, outExceptionString))
+        return false;
+
+    if (NSNumber *frameNumber = objectForKey<NSNumber>(options, frameIdKey)) {
```

Comment:
is it necessary to use `objectForKey` here if it passed the NSNumber.class validation check? i had considered using `objectForKey` as well in my scripting PR but thought it may be unnecessary.

Reply:
Yeah you are right, after validation we can assume the types here. I can change this.